src/App.js

```diff
@@ -1,19 +1,38 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
+// import ChatEntry from './components/ChatEntry';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+
+  const updateToggleLike = (updatedEntry) => {
+    const newChatData = chatData.map((entry) => {
+      if (entry.id === updatedEntry.id){
+        return updatedEntry;
+      } else {
+        return entry;
+      }
+    }); 
+    setChatData(newChatData)
+  }
```

Comment:
updateToggleLike executes when the like button on a ChatEntry component is clicked. We really only need to update the heart on a ChatEntry, not the entire message. How would you refactor updateToggleLike so that you aren't updating the entire ChatEntry object?

You can check out this [implementation](https://github.com/ada-c17/react-chatlog/blob/c17/solution/src/App.js#L22-L30) for an idea of how this refactor would look